designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsHtmlSnippet.scala

```diff
@@ -12,25 +14,68 @@ case class UsageStatisticsHtmlSnippet(value: String)
 
 object UsageStatisticsHtmlSnippet {
 
-  def prepareWhenEnabledReporting(config: UsageStatisticsReportsConfig): Option[UsageStatisticsHtmlSnippet] = {
+  private val knownDeploymentManagerTypes = Set("flinkStreaming", "lite-k8s", "lite-embedded")
+
+  private val streamingProcessingMode = "streaming"
+
+  private val knownProcessingModes = Set(streamingProcessingMode, "request-response")
+
+  // We aggregate custom deployment managers and processing modes as a "custom" to avoid leaking of internal, confidential data
+  private val aggregateForCustomValues = "custom"
+
+  def prepareWhenEnabledReporting(config: UsageStatisticsReportsConfig,
+                                  processingTypeStatistics: ProcessingTypeDataProvider[ProcessingTypeUsageStatistics]): Option[UsageStatisticsHtmlSnippet] = {
     if (config.enabled) {
-      val queryParams = prepareQueryParams(config)
+      val queryParams = prepareQueryParams(config, processingTypeStatistics.all)
       val url = prepareUrl(queryParams)
       Some(UsageStatisticsHtmlSnippet(s"""<img src="$url" alt="anonymous usage reporting" referrerpolicy="origin" hidden />"""))
     } else {
       None
     }
   }
 
-  private def prepareQueryParams(config: UsageStatisticsReportsConfig) = {
+  private[statistics] def prepareQueryParams(config: UsageStatisticsReportsConfig,
+                                             processingTypeStatisticsMap: Map[ProcessingType, ProcessingTypeUsageStatistics]): ListMap[String, String] = {
+    val deploymentManagerTypes = processingTypeStatisticsMap.values.map(_.deploymentManagerType).map {
+      case dm if knownDeploymentManagerTypes.contains(dm) => dm
+      case _ => aggregateForCustomValues
+    }
+    val dmParams = prepareValuesParams(deploymentManagerTypes, "dm")
+
+    val processingModes = processingTypeStatisticsMap.values.collect {
+      case ProcessingTypeUsageStatistics(_, Some(mode)) if knownProcessingModes.contains(mode) => mode
+      case ProcessingTypeUsageStatistics(_, Some(_)) => aggregateForCustomValues
```

Comment:
I think this should be default at the end, regardless of mode?

Reply:
:+1: changed